src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml.cpp

```diff
@@ -90,17 +90,29 @@ void EquationInputArea::AddNewEquation()
         return;
     }
 
-    m_lastLineColorIndex = (m_lastLineColorIndex + 1) % AvailableColors->Size;
-
     int colorIndex;
 
     if (m_accessibilitySettings->HighContrast)
     {
+        m_lastLineColorIndex = (m_lastLineColorIndex + 1) % AvailableColors->Size;
         colorIndex = m_lastLineColorIndex;
     }
     else
     {
-        colorIndex = colorAssignmentMapping[m_lastLineColorIndex];
+        std::array<bool, maxEquationSize> colorAssignmentUsed = { false, false, false, false, false, false, false,
+                                                                  false, false, false, false, false, false, false };
+        for (auto equation : Equations)
+        {
+            colorAssignmentUsed[equation->LineColorIndex] = true;
+        }
+
+        colorIndex = 0;
+        while (colorIndex < maxEquationSize - 1 && colorAssignmentUsed[colorAssignmentMapping[colorIndex]])
+        {
+            colorIndex++;
+        }
+
+        colorIndex = colorAssignmentMapping[colorIndex];
```

Comment:
it should not happen but let's still make sure that this function correctly support the cause where we didn't find a color available (so colorIndex is out of array). Suggestion: return the last color of the array

Reply:
In that case we will hit the colorIndex < maxEquationSize case of the while loop and always return the last color (already tested it)

Reply:
You are right, I missed the `<` instead of `<=`. Can you add a comment explaining it before someone tries to fix the code in a couple of years?

Reply:
You are right, I missed the `<` instead of `<=`. Can you add a comment explaining it before someone tries to fix the code in a couple of years?